### DISCUSSION TITLE
Is there a way to not make the modal fullscreen when using BottomSheetTextInput?

### DISCUSSION THREAD

**Discussion by glisom** [gorhom/react-native-bottom-sheet#1256] [Q&A]
When using the text input provided, my modal always goes fullscreen but would like to keep it at its current snap point.

**Comment by glisom** [comment]
Essentially if an input has autoFocus on the modal is opening 100% of the way regardless of snapPoints and I don't see anything in the docs on how to set the height when the keyboard is open when using BottomSheetModal.

**Reply by reillyjodonnell** [reply]
I am running into this problem right now. I'll update if I can figure it out.

**Reply by glisom** [reply]
Shoot, I totally forgot to update this. We were able to get around it, lemme see if I can find the fix.

**Reply by reillyjodonnell** [reply]
I'm using RN's KeyboardAvoidingView with a vertical offset. The code is like this:

```
<KeyboardAvoidingView
          enabled
          keyboardVerticalOffset={270}
          behavior={Platform.OS === 'ios' ? 'padding' : 'height'}>
          <TextInput
            onSubmitEditing={Keyboard.dismiss}
            textAlign="left"
            placeholder="Awesome 🎉"
            multiline
            style={styles.textInput}
          />
        </KeyboardAvoidingView>

```

**Reply by reillyjodonnell** [reply]
Here's a demo of the result
https://user-images.githubusercontent.com/65147216/221285855-8cdecdae-f02b-43a6-a656-46fafcd9f14a.mp4

**Reply by glisom** [reply]
Do you have your sheet change and other state change functions wrapped in useCallback? I think that is what fixed the weird behavior for us. And then on android, to prevent odd behavior with the keyboard, we added a 400 ms timeout before closing the modal after dismissing the keyboard

**Reply by ugoi** [reply]
> I'm using RN's KeyboardAvoidingView with a vertical offset. The code is like this:
> 
> ```
> <KeyboardAvoidingView
>           enabled
>           keyboardVerticalOffset={270}
>           behavior={Platform.OS === 'ios' ? 'padding' : 'height'}>
>           <TextInput
>             onSubmitEditing={Keyboard.dismiss}
>             textAlign="left"
>             placeholder="Awesome 🎉"
>             multiline
>             style={styles.textInput}
>           />
>         </KeyboardAvoidingView>
> ```

When I change the snap point from 90% to 80% i have to manually adjust the keyboardVerticalOffset otherwise the keyboard covers input.

**Comment by erquhart** [comment]
@gorhom I'm seeing this behavior still - snap points are ignored and the bottom sheet goes to 100% when the input is focused. Let me know if you have any workarounds.

**Reply by erquhart** [reply]
It looks like setting a `topInset` is one workaround. In my case, my max snap point was really just an indirect way of setting this value, so this should suffice.
